cpp/src/gandiva/function_registry.cc

```diff
@@ -109,11 +109,35 @@ arrow::Status FunctionRegistry::Register(const std::vector<NativeFunction>& func
   return Status::OK();
 }
 
+arrow::Status FunctionRegistry::Register(
+    NativeFunction func, void* c_function_ptr,
+    std::optional<FunctionHolderMaker> function_holder_maker) {
+  if (function_holder_maker.has_value()) {
+    // all signatures should have the same base name, use the first signature's base name
+    auto const& func_base_name = func.signatures().begin()->base_name();
+    ARROW_RETURN_NOT_OK(holder_maker_registry_.Register(
+        func_base_name, std::move(function_holder_maker.value())));
```

Comment:
Do we need this `std::move()`?
I think that a function call result is already rvalue.

Reply:
I think it should be `std::move(function_holder_maker).value()` to make it a rvalue

Reply:
Fat finger. I change it to `std::move(function_holder_maker).value()` now.